File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Go/c92c27565ddf4f5c66fa315e1c48ce9d.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n, k int
	fmt.Scan(&n, &k)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
		a[i] = -a[i]
	}
	sort.Ints(a)
	ans := 0
	res := 0
	for i := 0; i < n; i++ {
		res -= a[i]
		if res < 0 {
			ans += res
			break
		}
		if i != n-1 {
			ans += res
		}
	}
	if ans >= 0 {
		fmt.Println(ans)
		return
	}
	tmp := []int{res}
	for i := ans + 1; i < n; i++ {
		tmp = append(tmp, -a[i])
	}
	sort.Ints(tmp)
	w := 0
	for _, c := range tmp {
		ans += (w / (k + 1)) * c
		w++
	}
	fmt.Println(ans)
}

File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Python/c92c27565ddf4f5c66fa315e1c48ce9d.py
n, k = map(int, input().split())
a = list(map(int, input().split()))
a = [-x for x in a]
a.sort()
ans = 0
res = 0
for i in range(n):
    res -= a[i]
    if res < 0:
        j = i
        break
    if i != n-1:
        ans += res
if 'j' not in locals():
    print(ans)
    exit()
tmp = [res] + [-a[i] for i in range(j+1, n)]
tmp.sort()
w = 0
for c in tmp:
    ans += (w // (k+1)) * c
    w += 1
print(ans)
